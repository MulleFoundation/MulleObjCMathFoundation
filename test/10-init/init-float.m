#import <MulleObjCMathFoundation/MulleObjCMathFoundation.h>

#include <float.h>
#include <math.h>



static void   test_value( float v)
{
   NSNumber   *nr;
   NSString   *className;

   nr = [NSNumber numberWithFloat:v];
   if( ! nr)
   {
      printf( "%g fail\n", v);
      return;
   }
   className = NSStringFromClass( [nr class]);

   printf( "%s : %s\n", [[nr stringValue] UTF8String], [className UTF8String]);
}


int   main( int argc, char *argv[])
{
   test_value( 0.0);
   test_value( 1.0);
   test_value( 1848);
   // some big numbers that don't fit into the TPS
   test_value( (float) (INT_MIN / 32));
//   test_value( (double) NSIntegerMax); // this gives an overflow on conversion
   test_value( (float) (INT_MAX / 32));
   test_value( 18.48);
   test_value( INFINITY);
   test_value( NAN);

   return( 0);
}
