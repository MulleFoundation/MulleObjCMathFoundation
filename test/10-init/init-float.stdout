0 : _MulleObjCTaggedPointerIntegerNumber
1 : _MulleObjCTaggedPointerIntegerNumber
1848 : _MulleObjCTaggedPointerIntegerNumber
-67108864 : _MulleObjCTaggedPointerIntegerNumber
67108864 : _MulleObjCTaggedPointerIntegerNumber
18.479999542236328125 : _MulleObjCDoubleNumber
inf : _MulleObjCDoubleNumber
nan : _MulleObjCDoubleNumber
